### PR TITLE
Added support for syncing UI with remote cloud.

### PR DESCRIPTION
**Pull request by skazi0**
For testing with real mkcloud setup running on a remote host.
Steps:
Copy cloud.conf.json.template to cloud.conf.json.
Edit cloud.conf.json to match your setup.
Run `gulp` for default cloud or `gulp --sync-cloud=<keyfromconfig>`

### CODE REVIEW

**Comment by skazi0** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;OTD4FMsJ&#x2F;127-3-update-crowbar-ui-development-environment"> (3) Update Crowbar-UI Development environment</a>

File: cloud.conf.json.template [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "default": {
```
Comment (santiph) [inline cloud.conf.json.template]: Please, lets use some reference to cloud 6, 7 and cloud 7 crowbar init.
Also, we need 3 different ports for the ssh connection on each machine

File: gulp/rsync.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Modules required for this task
+var gulp = require('gulp'),
+    gutil = require('gulp-util'),
+    rsync = require('gulp-rsync'),
+    optional = require('optional'),
+    cloudconfigs = optional('./cloud.conf.json') || {};
+
+gulp.task('rsync', function () {
+    var cloudid = gutil.env.synccloud || 'default';
+    var cloudconfig = cloudconfigs[cloudid];
```
Comment (santiph) [inline gulp/rsync.js]: Please move `cloudid` and `cloudconfig` to line 7?
Also, camelcase

**Review by santiph — CHANGES_REQUESTED** [review]
Looks very good so far. I left a couple of comments to be addressed

File: gulp/rsync.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Modules required for this task
+var gulp = require('gulp'),
+    gutil = require('gulp-util'),
+    rsync = require('gulp-rsync'),
+    optional = require('optional'),
+    cloudconfigs = optional('./cloud.conf.json') || {};
+
+gulp.task('rsync', function () {
+    var cloudid = gutil.env.synccloud || 'default';
```
Comment (santiph) [inline gulp/rsync.js]: Please, use dashes on `sync-cloud` to match gulp's parameters convention

File: cloud.conf.json.template (hunk 1 above)
Comment (skazi0) [reply cloud.conf.json.template]: You can name your clouds any way you like. For multiple ports just add multiple cloud entries with the same host but different ports (and a different ID of course).

EDIT: I added more examples to the template config.

File: gulp/rsync.js (hunk 2 above)
Comment (skazi0) [reply gulp/rsync.js]: done.

File: gulp/rsync.js (hunk 3 above)
Comment (skazi0) [reply gulp/rsync.js]: done.

**Review by santiph — APPROVED** [review]

**Comment by rsalevsky** [comment]
Awesome that this got merged! I need to adapt the package in parallel to avoid conflicts.

**Pull request merged**